lib/util/git/auth.ts

```diff
@@ -115,33 +116,45 @@ function getAuthenticationRulesWithToken(
  */
 export function getAuthenticationRules(
   gitUrl: string,
+  hostType: string | undefined | null,
   token: string,
 ): AuthenticationRule[] {
   const authenticationRules = [];
   const hasUser = token.split(':').length > 1;
   const insteadUrl = parseGitUrl(gitUrl);
 
+  if (hostType === 'bitbucket-server') {
+    insteadUrl.source = 'bitbucket-server';
```

Comment:
How is this field used?

Reply:
https://github.com/IonicaBizau/git-url-parse/blob/master/lib/index.js#L304

Reply:
Already had to set it in `getHttpUrl()` previously: https://github.com/renovatebot/renovate/blob/main/lib/util/git/url.ts#L47 (#29527)